src/Disks/DiskLocal.cpp

```diff
@@ -337,6 +391,22 @@ void DiskLocal::sync(int fd) const
         throw Exception("Cannot fsync", ErrorCodes::CANNOT_FSYNC);
 }
 
+void DiskLocal::updateFromConfig(const Poco::Util::AbstractConfiguration & config, 
+                        const String & config_prefix,
+                        const Context & context)
+{
+    String new_disk_path;
+    UInt64 new_keep_free_space_bytes;
+    loadDiskLocalConfig(name, config, config_prefix, context, new_disk_path, new_keep_free_space_bytes);
+    if (disk_path != new_disk_path)
+        throw Exception("Disk path can't update from config " + name, ErrorCodes::UNKNOWN_ELEMENT_IN_CONFIG);
+
+    // Config of DiskLocal has been updated
+    if (keep_free_space_bytes != new_keep_free_space_bytes)
```

Comment:
@azat  The only parameter in `DiskLocal` is `keep_free_space_bytes`,  maybe the number of parameters in `DiskLocal` is too few to juding whether to reload config by version?

Reply:
@taiyang-li I was talking about more generic approach for config versioning:
- config should have some global version
- DiskLocal should cache it and call updateFromConfig only if the version of the config is newer then the cached version in DiskLocal

Reply:
Got it.  I will update this code.

Reply:
@taiyang-li  Also please take a look at https://github.com/ClickHouse/ClickHouse/pull/19526#discussion_r564302583 maybe you will prefer this way